### PR TITLE
Support unixfd transfer via dbus

### PR DESCRIPTION
**Pull request by wdouglass**
Also, add tests for same

### CODE REVIEW

**Comment by wdouglass** [comment]
Hmmm -- is there a way to notify codecov of the forking tests? looks like some coverage isn't getting measured

**Comment by wdouglass** [comment]
hmmm -- looks like coverage is still missing a line in xml.py -- i'm not sure why at all

I didn't touch the xml tests, i don't know what changed here

I'd love a second set of eyes :)

**Comment by tjoslin** [comment]
@poncovka - this is the "part 3 of 3" for Woodrow's unixfd support feature addition (following on from #70 and #72). Could you let us know if there's more work you'd like to see prior to a merge?

File: Makefile [hunk 1]
```diff
@@ -52,8 +52,13 @@ check:
 .PHONY: test
 test:
 	@echo "*** Running unittests with $(COVERAGE) ***"
-	PYTHONPATH=. $(COVERAGE) run --branch -m unittest discover -v -s tests/
-	$(COVERAGE) report -m --include="dasbus/*" | tee coverage-report.log
+	echo "[run]" > .coveragerc
+	echo "concurrency = multiprocessing,thread" >> .coveragerc
+	echo "branch = true" >> .coveragerc
+	echo "parallel = true" >> .coveragerc
```
[the remark this answers is not in the document]
Comment (wdouglass) [reply Makefile:58]: done, #82

File: dasbus/client/handler.py [hunk 2]
```diff
@@ -54,36 +56,76 @@ def sync_call(cls, connection, service_name, object_path, interface_name,
...
-            timeout,
-            None
-        )
+        if parameters:
+            params, fdlist = variant_replace_handles_with_fdlist_indices(
+                parameters)
+        else:
+            fdlist = []
+        if len(fdlist) > 0:
+            ret =  connection.call_with_unix_fd_list_sync(
```
[the remark this answers is not in the document]
Comment (wdouglass) [reply dasbus/client/handler.py:65]: Ok; i'll mark this as a draft and refactor to seperate out the unix-only calls.

[the remark this answers is not in the document]
Comment (wdouglass) [reply dasbus/client/handler.py]: I've seperated them out, see test_dbus.py for an example of use

I'd like to have unixfd support on by default for systems that support them, but i haven't figured out how to make that decision yet. libdbus has this call
https://dbus.freedesktop.org/doc/api/html/group__DBusConnection.html#ga3e41509b3afdbc22872bacc5754e85c2

i haven't found an analogue for this call in pygobject (yet...)

**Comment by poncovka** [comment]
> that codecov/project is failing because there's a bunch of code in test_dbus.py that runs in a seperate subprocess that's not being counted by codecov. is there a way to mark that as external to codecov?

Let me check tomorrow. I will probably merge the current version and do a clean up in additional pull requests. I would like to include this feature in the next release and release it soon if possible.

**Comment by wdouglass** [comment]
That's great news! thank you very much

**Review by poncovka — APPROVED** [review]
Looks good to me. Thank you!

**Pull request merged**